nightfall-deployer/src/contract-setup.mjs

```diff
@@ -4,96 +4,62 @@ address of the contract that holds global state (State.sol)
-          .encodeABI(),
-        proposerContractInstance.options.address,
-      );
-      await Web3.submitRawTransaction(
-        shieldContractInstance.methods.setStateContract(stateInstance.options.address).encodeABI(),
-        shieldContractInstance.options.address,
-      );
-      await Web3.submitRawTransaction(
-        challengesContractInstance.methods
-          .setStateContract(stateInstance.options.address)
-          .encodeABI(),
-        challengesContractInstance.options.address,
-      );
-    } catch (err) {
-      if (err.message.includes('Transaction has been reverted by the EVM'))
-        logger.warn(
-          'Writing contract addresses to the State contract failed. This is probably because they are aready set. Did you already run deployer?',
-        );
-      else throw new Error(err);
-    }
+  const simpleMultiSigAddress = (await waitForContract('SimpleMultiSig')).options.address;
 
-    // our last action as the deployer is to hand off our onlyOwner powers to the
-    // multisig contract
-    await Web3.submitRawTransaction(
-      shieldContractInstance.methods.transferOwnership(simpleMultiSigAddress).encodeABI(),
-      shieldContractInstance.options.address,
-    );
-    await Web3.submitRawTransaction(
-      stateContractInstance.methods.transferOwnership(simpleMultiSigAddress).encodeABI(),
-      stateContractInstance.options.address,
-    );
-    await Web3.submitRawTransaction(
-      proposerContractInstance.methods.transferOwnership(simpleMultiSigAddress).encodeABI(),
-      proposerContractInstance.options.address,
-    );
-    return Web3.submitRawTransaction(
-      challengesContractInstance.methods.transferOwnership(simpleMultiSigAddress).encodeABI(),
-      challengesContractInstance.options.address,
+  const contracts = {
+    proposers,
+    shield,
+    challenges,
+  };
+
+  for await (const contractName of ['proposers', 'shield', 'challenges']) {
+    const setStateContract = contracts[contractName].methods.setStateContract(
+      stateInstance.options.address,
     );
-  }
 
-  try {
-    // the following code runs the registrations in parallel
-    await Promise.all([
-      proposerContractInstance.methods.setStateContract(stateInstance.options.address).send(),
-      shieldContractInstance.methods.setStateContract(stateInstance.options.address).send(),
-      challengesContractInstance.methods.setStateContract(stateInstance.options.address).send(),
-    ]);
-  } catch (err) {
-    if (err.message.includes('Transaction has been reverted by the EVM'))
-      logger.warn(
-        'Writing contract addresses to the State contract failed. This is probably because they are aready set. Did you already run deployer?',
+    const transferOwnership =
+      contracts[contractName].methods.transferOwnership(simpleMultiSigAddress);
+
+    if (!config.ETH_PRIVATE_KEY) {
+      setStateContract.send();
+      transferOwnership.send();
+    } else {
+      tx.data = setStateContract.encodeABI();
```

Comment:
What if instead of this code
```
      tx.data = setStateContract.encodeABI();
      let signed = await web3.eth.accounts.signTransaction(
        { ...tx, to: contracts[contractName].options.address },
        config.ETH_PRIVATE_KEY,
      );
      await web3.eth.sendSignedTransaction(signed.rawTransaction);
```
we use the previous function that do the same code? 

```
      await Web3.submitRawTransaction(
        setStateContract.encodeABI(),
        contracts[contractName].options.address,
      );
```

Reply:
indeed this seems to do the same thing, I don't know how I arrived on the top code, I'll try to refactor and see if there was any side effect I'm not remembering now